src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -175,17 +199,19 @@ def _read_tool_result(result_dir, batch_dir, kbase_collection, load_ver, tool_fi
     # retrieve and process the genome metadata file
     metadata_file = os.path.join(batch_result_dir, loader_common_names.GENOME_METADATA_FILE)
     try:
-        meta_df = pd.read_csv(metadata_file, sep='\t', names=['tool_identifier', 'genome_id', 'source_file_path'])
+        meta_df = pd.read_csv(metadata_file, sep='\t')
     except Exception as e:
         raise ValueError('Unable to retrieve the genome metadata file') from e
+    
     tool_genome_map = dict(zip(meta_df.tool_identifier, meta_df.genome_id))
+    tool_upa_map = _create_tool_upa_map(meta_df.tool_identifier, meta_df.source_dir, meta_df.meta_filename)
 
     tool_file = os.path.join(result_dir, str(batch_dir), tool_file_name)
     docs = dict()
     if os.path.exists(tool_file):
         df = _read_tsv_as_df(tool_file, features, genome_id_col=genome_id_col)
         docs = df.apply(_row_to_doc, args=(kbase_collection, load_ver, features, tool_genome_map,
-                                           genome_id_col, prefix), axis=1).to_list()
+                                           tool_upa_map, genome_id_col, prefix), axis=1).to_list()
```

Comment:
The upa data should only be added to the genome attributes output, e.g. for the function call `_process_genome_attri_tools` at the bottom of the file.

With the current implementation those source meta files are being read twice, once for gtdb_tk and once for checkm2, so the UPA information should be added in that function after the document merge operation, where the tool identifier is irrelevant.

Reply:
Do you think it makes more sense to keep a global var i.e. `HAS_READ_META` and only call `_create_tool_upa_map` when `HAS_READ_META` is `False`. If the logic gets moved to `_process_genome_attri_tools`, the logic then needs to figure out batch_dir and then `_key` for each genome.

Reply:
Updated!

Reply:
:+1: